### PR TITLE
Jenkins test for PetSc

### PR DESCRIPTION
**Pull request by bilke**
TODO: 

- [x] `OGS_USE_PETSC=ON`
- ~~~`OGS_USE_MPI=ON` --> does not compile~~~
- [x] Check tests

### CODE REVIEW

File: scripts/jenkins/gcc-dynamic.groovy [hunk 1]
```diff
@@ -3,27 +3,45 @@ def defaultCMakeOptions =
...
 
 def configure = new ogs.configure()
 def build = new ogs.build()
 def post = new ogs.post()
 def helper = new ogs.helper()
 
-stage('Configure (Linux-Docker-Dynamic)') {
+stage('Configure (envinf1)') {
     configure.linux(cmakeOptions: defaultCMakeOptions, env: 'envinf1/cli.sh', script: this)
+    configure.linux(cmakeOptions: defaultCMakeOptions + '-DOGS_USE_MPI=ON',
```
Comment (wenqing) [inline scripts/jenkins/gcc-dynamic.groovy]: As far as I knew that there is no stand alone implementation and test for  option of OGS_USE_MPI. OGS_USE_MPI should be on only when OGS_USE_PETSC is ON.

**Comment by bilke** [comment]
@wenqing: [Some tests fail](https://jenkins.opengeosys.org/job/ufz/job/ogs/job/PR-1718/2/execution/node/182/log/), please have a look.

**Comment by wenqing** [comment]
@bilke  I am checking now.

**Comment by wenqing** [comment]
@bilke When I compiled it on frontend of EVE, I got an error to down load eigen as
```
CMake Error at Eigen-stamp/download-Eigen.cmake:27 (message):
  error: downloading 'http://bitbucket.org/eigen/eigen/get/3.2.9.tar.gz'
  failed
    status_code: 1
    status_string: "Unsupported protocol"
    log:   Trying 104.192.143.3...
  Connected to bitbucket.org (104.192.143.3) port 80 (#0)
...
```
Have you ever encountered such problem?

**Comment by bilke** [comment]
@wenqing Should be fixed with #1719.

**Comment by wenqing** [comment]
@bilke  Two more errors on eve:
1. binary files of vtk are downloaded in ``.../exe_petsc/External/vtk/src/vtk`` but cmake try to find CMakeLists.txt in the directory.
```
loading initial cache file /home/wwang/code/ogs6/exe_petsc/External/vtk/tmp/vtk-cache-Release.cmake
CMake Error: The source directory "/home/wwang/code/ogs6/exe_petsc/External/vtk/src/vtk" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
make[2]: *** [External/vtk/src/vtk-stamp/vtk-configure] Error 1
make[1]: *** [CMakeFiles/vtk.dir/all] Error 2
```
2. Several similar errors like:
 ```
[  4%] Building CXX object ThirdParty/tetgen/CMakeFiles/tet.dir/predicates.cxx.o
/tmp/ccqObPKQ.s: Assembler messages:
/tmp/ccqObPKQ.s:349: Error: no such instruction: `vextracti128 $0x1,%ymm0,24(%rdi)'
```

Could you please take a look?

**Comment by bilke** [comment]
There seems to be something wrong with your config. Please see [gcc-dynamic.groovy](https://github.com/bilke/ogs/blob/3b2f83288bda32bf99fa5d71d1299c149d8694ca/scripts/jenkins/gcc-dynamic.groovy) for the config used by Jenkins. Basically it does the following:

```bash
source scripts/env/envinf1/petsc.sh # Loads modules
cmake -DCMAKE_BUILD_TYPE=Release -DOGS_LIB_BOOST=System -DOGS_LIB_VTK=System -DBUILD_SHARED_LIBS=ON -DOGS_BUILD_UTILS=ON -DOGS_USE_PETSC=ON ../ogs # configures
make tests ctest # builds
```

**Comment by wenqing** [comment]
@bilke Found the reason for the compilation errors: OGS_LIB_VTK is set to Local. When  OGS_LIB_VTK  is set to System, the only working module is vtk/6.3.0_openmpi-1.8.4-noqt-1. Thanks.

**Comment by wenqing** [comment]
@bilke  Just sent a PR to your branch of this PR about the test error.

**Comment by bilke** [comment]
@wenqing Thanks!

@endJunction Something is wrong with your machine and docker:

```
java.io.IOException: Failed to run image 'ogs6/gcc-gui:latest'.
Error: docker: Error response from daemon: oci runtime error: container_linux.go:247:
  starting container process caused "process_linux.go:334: running prestart hook 0
  caused \"fork/exec /usr/bin/dockerd (deleted): no such file or directory\"".
```

This fails also in `master`. Can you please manually run:

```
docker run -it ogs6/gcc-gui:latest bash
```

**Pull request merged**